packages/contracts-bedrock/test/LivenessModule.t.sol

```diff
@@ -0,0 +1,82 @@
+
+    event SignersRecorded(bytes32 indexed txHash, address[] signers);
+
+    LivenessModule livenessModule;
+    LivenessGuard livenessGuard;
+    SafeInstance safeInstance;
+
+    function setUp() public {
+        // Create a Safe with 10 owners
+        (, uint256[] memory keys) = makeAddrsAndKeys(10);
+        safeInstance = _setupSafe(keys, 8);
+        livenessGuard = new LivenessGuard(safeInstance.safe);
+        livenessModule = new LivenessModule({
+            _safe: safeInstance.safe,
+            _livenessGuard: livenessGuard,
+            _livenessInterval: 30 days,
+            _minOwners: 6,
+            _fallbackOwner: makeAddr("fallbackOwner")
+        });
+        safeInstance.enableModule(address(livenessModule));
+        safeInstance.setGuard(address(livenessGuard));
+    }
+}
+
+contract LivenessModule_Get75PercentThreshold_Test is LivnessModule_TestInit {
+    /// @dev check the return values of the get75PercentThreshold function against manually
+    ///      calculated values.
+    function test_get75PercentThreshold_Works() external {
+        assertEq(livenessModule.get75PercentThreshold(20), 15);
+        assertEq(livenessModule.get75PercentThreshold(19), 15);
+        assertEq(livenessModule.get75PercentThreshold(18), 14);
+        assertEq(livenessModule.get75PercentThreshold(17), 13);
+        assertEq(livenessModule.get75PercentThreshold(16), 12);
+        assertEq(livenessModule.get75PercentThreshold(15), 12);
+        assertEq(livenessModule.get75PercentThreshold(14), 11);
+        assertEq(livenessModule.get75PercentThreshold(13), 10);
+        assertEq(livenessModule.get75PercentThreshold(12), 9);
+        assertEq(livenessModule.get75PercentThreshold(11), 9);
+        assertEq(livenessModule.get75PercentThreshold(10), 8);
+        assertEq(livenessModule.get75PercentThreshold(9), 7);
+        assertEq(livenessModule.get75PercentThreshold(8), 6);
+        assertEq(livenessModule.get75PercentThreshold(7), 6);
+        assertEq(livenessModule.get75PercentThreshold(6), 5);
+        assertEq(livenessModule.get75PercentThreshold(5), 4);
+        assertEq(livenessModule.get75PercentThreshold(4), 3);
+        assertEq(livenessModule.get75PercentThreshold(3), 3);
+        assertEq(livenessModule.get75PercentThreshold(2), 2);
+        assertEq(livenessModule.get75PercentThreshold(1), 1);
+    }
+}
+
+contract LivenessModule_RemoveOwner_Test is LivnessModule_TestInit {
+    function test_removeOwner_oneOwner_succeeds() external {
+        uint256 ownersBefore = safeInstance.owners.length;
+        vm.warp(block.timestamp + 30 days);
+        livenessModule.removeOwner(safeInstance.owners[0]);
+        assertEq(safeInstance.safe.getOwners().length, ownersBefore - 1);
+    }
+
+    function test_removeOwner_allOwners_succeeds() external {
+        vm.warp(block.timestamp + 30 days);
+        // The safe is initialized with 10 owners, so we need to remove 3 to get below the minOwners threshold
```

Comment:
Can we call for the number of owners and then compute the number to remove to decouple from out of sight logic?